[Workers] Update Fauna tutorial

yusukebe opened the pull request:
Updated "Create a serverless, globally distributed REST API with Fauna" tutorial:

- Use Hono as a web framework instead of Worktop, as Worktop is not currently being maintained.
- Use C3 for initializing a Worker project.
- Use TypeScript.
- Rewrite some phrases.

### Note

- Should we remove `?utm_source=Cloudflare&utm_medium=referral&utm_campaign=Q4_CF_2021` in links to Fauna.
- Reference-style links does not work in `aside`. Perhaps, we don't have to use Reference-style.
- Currently, the repo hosting complete code is [my repo](https://github.com/yusukebe/fauna-workers). Should I make it in Cloudflare org?
- [Fauna Workers quickstart](https://github.com/fauna-labs/fauna-workers) appears to be outdated.

yusukebe commented:
Hi @deadlypants1973 !

Since this PR was merged, we can initialize a Hono project more easily with C3. I'll need to rewrite this tutorial accordingly. I'll tackle it next week. Thank you!

https://github.com/cloudflare/workers-sdk/pull/3552

deadlypants1973 commented on content/workers/tutorials/store-data-with-fauna/index.md:
````diff
@@ -115,168 +134,187 @@ When prompted, paste the Fauna server secret you obtained earlier.
 ```
 
-The `FAUNA_SECRET` environment variable is injected into your application automatically at runtime. Workers run on a custom JavaScript runtime instead of Node.js, so you do not need to use `process.env` to access your environment variables.
+You can access `FAUNA_SECRET` environment variables from `c.env.FAUNA_SECRET`. Workers run on a custom JavaScript runtime instead of Node.js, so you can't use `process.env` to access your environment variables.
````
```suggestion
You can access `FAUNA_SECRET` environment variable from `c.env.FAUNA_SECRET`. Workers run on a [custom JavaScript runtime](https://developers.cloudflare.com/workers/runtime-apis/nodejs/) instead of Node.js, so you cannot use `process.env` to access your environment variables.
```

deadlypants1973 commented on content/workers/tutorials/store-data-with-fauna/index.md:
````diff
@@ -345,109 +382,118 @@ Examining the route you create, when the query is successful, the ID of the newl
+
+```sh
+---
+header: Deploy your Worker
+---
+$ npm run start
````
This is saying to use Wrangler dev, but the header is deploy Worker and the command is not dev or deploy?

yusukebe replied:
The header should be "Develop", not "Deploy".

deadlypants1973 commented on content/workers/tutorials/store-data-with-fauna/index.md:
```diff
@@ -345,109 +382,118 @@ Examining the route you create, when the query is successful, the ID of the newl
 ---
-header: Testing your Worker locally
+header: Deploy your Worker
 ---
-wrangler dev
+$ yarn start
```
same comment as above

deadlypants1973 commented on content/workers/tutorials/store-data-with-fauna/index.md:
```diff
@@ -509,15 +555,32 @@ header: Read product response
+header: Deploy your Worker
 ---
-$ wrangler publish
+$ npm run deploy
```
```suggestion
$ npx wrangler deploy
```

deadlypants1973 commented on content/workers/tutorials/store-data-with-fauna/index.md:
```diff
@@ -907,10 +773,7 @@ If you would like to speak directly with a Fauna expert about building your appl
-[worktop]: https://github.com/lukeed/worktop
 [wrangler-dev]: /workers/wrangler/commands/#dev
-[wrangler-init]: /workers/wrangler/commands/#init
 [wrangler-publish]: /workers/wrangler/commands/#publish
```
```suggestion
[wrangler-deploy]: /workers/wrangler/commands/#deploy
```

deadlypants1973 commented on content/workers/tutorials/store-data-with-fauna/index.md:
```diff
@@ -662,215 +712,36 @@ header: Update product response
-## Cleaning up
+## Clean up
 
 To remove the resources you create in this tutorial, delete your Worker in the Cloudflare dashboard > **Workers & Pages** > select your Worker > **Manage Service** > **Delete**:
```
```suggestion
To remove the resources you create in this tutorial, delete your Worker in the Cloudflare dashboard > **Workers & Pages** > select your Worker > **Manage application** > **Delete**:
```

deadlypants1973 reviewed (commented):
PCX

yusukebe commented:
Wow, just a few days ago, the new version v10 of Fauna FQL was released as GA. The tutorial was originally written for v4, so I need to completely revise both the article and the sample project. I'll need some time for this.

yusukebe commented:
Ooops.

**I’m really sorry**. I mistakenly pushed everyone's commits that were unrelated to this PR with the review request and tags. I’m really sorry.

yusukebe commented:
Hi @deadlypants1973!

I made a big mistake. But I've done revising this tutorial. As mentioned, the API and dashboard of Fauna have undergone major changes. I've updated the example code and removed some verbose descriptions. Please review it! Thanks!

deadlypants1973 commented on content/workers/tutorials/store-data-with-fauna/index.md:
````diff
@@ -34,288 +30,293 @@ Fauna is a document-based database with a flexible schema. This allows you to de
 ```
 
-To review what a document looks like, navigate to the **Shell** tab in the Fauna dashboard and run the following query:
+To review what a document looks like, run the following query in the Fauna dashboard > Explorer > Region name > Database name like a `cloudflare_rest_api` > the **SHELL** window:
````
Is this asking the user to create a new database? This step is unclear.

deadlypants1973 commented on content/workers/tutorials/store-data-with-fauna/index.md:
```diff
@@ -34,288 +30,293 @@ Fauna is a document-based database with a flexible schema. This allows you to de
+### Create product documents
 
-Add your first Worktop route to the `index.js` file. This route accepts `POST` requests to the `/products` endpoint:
+Add your first Hono handler to the `src/index.ts` file. This route accepts `POST` requests to the `/products` endpoint:
```
Does adding a Hono handler create product documents?

yusukebe replied:
The handler `POST /products` which is described here creates product documents.

yusukebe replied:
Perhaps the naming is unclear. We call this a `handler` and also a `route`.

yusukebe commented on content/workers/tutorials/store-data-with-fauna/index.md:
```diff
@@ -34,288 +30,293 @@ Fauna is a document-based database with a flexible schema. This allows you to de
 - **quantity** A non-negative integer that represents how many items of a particular product there are in the inventory.
 
-Documents are stored in the **Products** [collection][fauna-collections]. Collections in document databases are groups of related documents.
+Documents are stored in the **Products** [collection](https://docs.fauna.com/fauna/current/reference/schema_entities/collection/). Collections in document databases are groups of related documents.
```
You are right. I'll fix it.

yusukebe commented on content/workers/tutorials/store-data-with-fauna/index.md:
```diff
@@ -34,288 +30,293 @@ Fauna is a document-based database with a flexible schema. This allows you to de
+
+{{</Aside>}}
+
+This route applied an FQL query in the `fql` function that creates a new document in the **Product** collection:
```
Yes. Should be `Products`.
